### PR TITLE
[FrameworkBundle] Change the default value of cookie_httponly

### PR DESCRIPTION
**Pull request by jderusse**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15303 |
| License | MIT |
| Doc PR | symfony/symfony-docs#5561 |

### CODE REVIEW

File: UPGRADE-3.0.md [hunk 1]
```diff
@@ -468,6 +468,8 @@ UPGRADE FROM 2.x to 3.0
    interface.
    The `security.csrf.token_manager` should be used instead.
 
+ * The default value of the parameter `cookie_httponly` is now `true`.
```
Comment (javiereguiluz) [inline UPGRADE-3.0.md]: I don't know if the UPGRADE guide is the best place for this, but should we also add a brief phrase exaplining the consequences of this change?

**Comment by jderusse** [comment]
It's a BC break because the default value changed. But it's for the own good of the application. It could be a issue on an application which retrieve the cookie value in javascript on purpose.

BTW I don't see how to notify the developper: I'm :-1: for adding a deprecated notice when the parameter is not overrided.

Other question, should I add a tests who asserts defaults values?

**Comment by javiereguiluz** [comment]
@jderusse I wasn't thinking about a deprecation notice. If I understand this change right, any Symfony app that uses JavaScript to read cookies values will stop working. I was wondering if we could say that explicitly somewhere.

**Comment by jderusse** [comment]
@javiereguiluz You're right, thats why it's a BC break.

Adding a deprecated notice when the parameter is not overrided will force everybody to explicitly sets this optional parameter :(

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by xabbuh** [comment]
If this should be merged into the `2.8` branch, the upgrade docs for 2.8 should be updated.

**Comment by iltar** [comment]
I think this change is worth the (minor) BC break it provides pre-3.0.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -340,7 +340,7 @@ private function addSessionSection(ArrayNodeDefinition $rootNode)
                         ->scalarNode('cookie_path')->end()
                         ->scalarNode('cookie_domain')->end()
                         ->booleanNode('cookie_secure')->end()
-                        ->booleanNode('cookie_httponly')->end()
+                        ->booleanNode('cookie_httponly')->defaultValue(true)->end()
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: please use ``defaultTrue()```instead for consistency

File: UPGRADE-3.0.md [hunk 3]
```diff
@@ -468,6 +468,18 @@ UPGRADE FROM 2.x to 3.0
    interface.
    The `security.csrf.token_manager` should be used instead.
 
+ * The default value of the parameter `session`.`cookie_httponly` is now `true`.
+   It prevents scripting languages, such as JavaScript to access the cookie,
+   which help to reduce identity theft through XSS attacks. If your
+   application need to access the session's cookie override this parameter:
```
Comment (stof) [inline UPGRADE-3.0.md]: ``the session cookie``

**Comment by stof** [comment]
@javiereguiluz this will only break things if you use Javascript to read the session cookie.

and :+1: for this change. It makes the session cookie safer by default

**Comment by jderusse** [comment]
I set the description in both `UPGRADE-2.8.md` and `UPGRADE-3.0.md`. Is it ok ?

**Comment by stof** [comment]
if the change is done in 2.8, there is nothing to add in ``UPGRADE-3.0.md``, as upgrading from 2.8 to 3.0 won't change anything

**Comment by jderusse** [comment]
fixed

File: UPGRADE-2.8.md [hunk 4]
```diff
@@ -136,3 +136,17 @@ DependencyInjection
...
    </services>
    ```
+
+FrameworkBundle
+---------------
+
+ * The default value of the parameter `session`.`cookie_httponly` is now `true`.
+   It prevents scripting languages, such as JavaScript to access the cookie,
+   which help to reduce identity theft through XSS attacks. If your
+   application need to access the session cookie override this parameter:
```
Comment (xabbuh) [inline UPGRADE-2.8.md]: needs

Comment (xabbuh) [inline UPGRADE-2.8.md]: And there should be a coma before override.

**Comment by fabpot** [comment]
Thank you @jderusse.

**Pull request merged**